### PR TITLE
Add DeviceSet to Cluster

### PR DESCRIPTION
**Pull request by aaroey**
Add DeviceSet to Cluster so we can access memory allocators when running grappler optimizations.
This is required by the TensorRT integration with TF.

### CODE REVIEW

**Comment by aaroey** [comment]
@samikama

**Review by benoitsteiner — CHANGES_REQUESTED** [review]

File: tensorflow/core/grappler/clusters/BUILD [hunk 1]
```diff
@@ -56,6 +56,7 @@ cc_library(
     ],
     visibility = ["//visibility:public"],
     deps = [
+        "//tensorflow/core:core_cpu_lib",
```
[the remark this answers is not in the document]
Comment (aaroey) [reply tensorflow/core/grappler/clusters/BUILD]: Done.

File: tensorflow/core/grappler/clusters/cluster.h [hunk 2]
```diff
@@ -88,6 +89,10 @@ class Cluster {
     return devices_;
   }
 
+  const DeviceSet* GetDeviceSet() const {
```
[the remark this answers is not in the document]
Comment (aaroey) [reply tensorflow/core/grappler/clusters/cluster.h]: Done.

**Review by aaroey — COMMENTED** [review]
Done, thanks for the review.

**Review by benoitsteiner — APPROVED** [review]

**Pull request merged**